Wikipedia/Code/SessionSingleton.m

```diff
@@ -201,6 +201,14 @@ - (void)fetchFinished:(id)sender
                 [self cloneSessionCookies];
             }
             break;
+            case FETCH_FINAL_STATUS_FAILED: {
+                // If autoLogin fails the credentials need to be cleared out if they're no longer valid so the
+                // user has an indication that they're no longer logged in.
+                if (error.code == LOGIN_ERROR_PASSWORD_WRONG){
```

Comment:
@coreyfloyd 
> Also we should never be worried about erasing credentials that do not work. Obviously the app can't use them to login so what good are they?

Hmm... but we don't store bad credentials, right? So we should never even be trying to autologin with anything but credentials which worked the only way a credential goes from `good` to `bad` is the user's password changed or they were blocked. Does this make sense?